WooCommerce/WooCommerceTests/POS/ViewModels/TotalsViewModelTests.swift

```diff
@@ -10,6 +28,43 @@ final class TotalsViewModelTests: XCTestCase {
     func test_formattedOrderTotalPrice() {}
     func test_formattedOrderTotalTaxPrice() {}
     func test_areAmountsFullyCalculated() {}
-    func test_clearOrder() {}
+    func test_clearOrder() {
+        // When
+        sut.clearOrder()
+
+        // Then
+        XCTAssertNil(sut.order)
+    }
     func test_setOrder() {}
+    func test_startNewTransaction_after_collecting_payment() async throws {
+        // Given
+        let paymentState: TotalsViewModel.PaymentState = .acceptingCard
+        let item = Self.makeItem()
+
+        await sut.syncOrder(for: [CartItem(id: UUID(), item: item, quantity: 1)], allItems: [item])
+        XCTAssertNotNil(sut.order)
+        let order: Order = orderService.order(from: sut.order!)
+
+        // When
+        _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
```

Comment:
We need to set `paymentEvent` on this mocked `cardPresentPaymentService`, rather than call `collectPayment` in the test. We can assert that `collectPayment` is called but check how `cardPresentPaymentInlineMessage` changes based on emitted `paymentEvent`.